code/modules/detectivework/scanner.dm

```diff
@@ -165,3 +166,29 @@
 
 /proc/get_timestamp()
 	return time2text(world.time + 432000, ":ss")
+
+/obj/item/device/detective_scanner/proc/can_use(mob/user)
+	if(user && ismob(user))
+		if(!user.incapacitated())
+			return TRUE
+	return FALSE
+	
+/obj/item/device/detective_scanner/AltClick(mob/living/user)
+	if(!can_use(user))
+		return
+	if(!LAZYLEN(log) || scanning)
+		to_chat(user, "<span class='notice'>Cannot clear logs, the scanner has no logs or is in use.</span>")
+		return
+	to_chat(user, "<span class='notice'>The scanner logs are cleared.</span>")
+	log = list()
+
+/obj/item/device/detective_scanner/verb/displayResults()

```

Comment:
Can you please use action buttons instead?

Reply:
I prefer that action buttons are mainly combat only but if you insist.

Reply:
Oh maybe you can help pkp with the ling action buttons then :) 

On TG the default is always buttons regardless of combat usage, you should only use verbs if you have to.

Reply:
Adding verbs generally isn't allowed (admin stuff being the exception, but even those use /proc shennanigans), we should write that somewhere